tpp-app/src/services/CalculationService.js

```diff
@@ -0,0 +1,162 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import {FLOW_LEVEL, KEYS} from "./utils/constants";
+import subDays from 'date-fns/subDays';
+import addDays from 'date-fns/addDays';
+import {getCalendarByYear, getDaysDiffInclusive, getSymptomsFromCalendar} from "./utils/helpers";
+import CycleService from "./cycle/CycleService";
+import { GETStoredYears } from "./utils/helpers";
+
+
+/**
+ * Returns Array of the Symptoms of the last three days
+ * @return {Array} of size 3
+ */
+const GETLastThreeDaysSymptoms = async () => {
+    const today = new Date();
+    const yesterday = subDays(today, 1);
+    const twoDaysEarlier = subDays(today, 2);
+
+    const calendar = await getCalendarByYear(today.getFullYear());
+
+    const todaySymptoms = await getSymptomsFromCalendar(
+        calendar,
+        today.getDay(),
+        today.getMonth() + 1,
+        today.getFullYear()
+    );
+
+    const yesterdaySymptoms = await getSymptomsFromCalendar(
+        calendar,
+        yesterday.getDay(),
+        yesterday.getMonth() + 1,
+        yesterday.getFullYear()
+    );
+
+    const twoDaysEarlierSymptoms = await getSymptomsFromCalendar(
+        calendar,
+        twoDaysEarlier.getDay(),
+        twoDaysEarlier.getMonth() + 1,
+        twoDaysEarlier.getFullYear()
+    );
+
+    return [twoDaysEarlierSymptoms, yesterdaySymptoms, todaySymptoms];
+};
+
+/**
+ * Returns True if today is the day a period is over, False otherwise i.e. matches period patten of X _ _
+ * @return {boolean} Resolves into a boolean indicating if period is over today
+ */
+export const isPeriodOver = async () => {
+    const lastThreeDaysSymptoms = await GETLastThreeDaysSymptoms();
+    const twoDaysEarlierSymptoms = lastThreeDaysSymptoms[0];
+    const yesterdaySymptoms = lastThreeDaysSymptoms[1];
+    const todaySymptoms = lastThreeDaysSymptoms[2];
+
+    const todayNoPeriod = (todaySymptoms.flow == null || todaySymptoms.flow === FLOW_LEVEL.NONE);
+    const yesterdayNoPeriod = (yesterdaySymptoms.flow == null || yesterdaySymptoms.flow === FLOW_LEVEL.NONE);
+    const twoDaysEarlierPeriod = (twoDaysEarlierSymptoms.flow != null && twoDaysEarlierSymptoms.flow !== FLOW_LEVEL.NONE);
```

Comment:
Is there a reason why we're not using triple comparators `===`? Alternatively for checking null you can use `!`:
```suggestion
    const todayNoPeriod = (!todaySymptoms.flow || todaySymptoms.flow === FLOW_LEVEL.NONE);
    const yesterdayNoPeriod = (!yesterdaySymptoms.flow || yesterdaySymptoms.flow === FLOW_LEVEL.NONE);
    const twoDaysEarlierPeriod = (!twoDaysEarlierSymptoms.flow && twoDaysEarlierSymptoms.flow !== FLOW_LEVEL.NONE);
```